packages/azure/data_stream/signinlogs/elasticsearch/ingest_pipeline/default.yml

```
@@ -130,12 +131,26 @@ processors:
   - remove:
       field: azure.signinlogs.level
       ignore_missing: true
+  - convert:
+      field: azure.signinlogs.duration_ms
+      target_field: event.duration
+      type: long
+      ignore_missing: true
+      ignore_failure: true
+      if: "ctx.azure?.signinlogs?.duration_ms != null && ctx.azure?.signinlogs?.duration_ms instanceof String"
```

```suggestion
      if: "ctx.azure?.signinlogs?.duration_ms != null && ctx.azure.signinlogs.duration_ms instanceof String"
```
We already know `ctx.azure.signinlogs` is not null from first condition.